### PR TITLE
Improving accu/laptop prediction/ac

### PR DESCRIPTION
**Pull request by nikhil1205-ai**
# Improve Model Accuracy

## Overview
This project focuses on analyzing laptop prices using a dataset of 1,300 laptop models. Through exploratory data analysis and the application of machine learning algorithms, it aims to uncover patterns and predict prices based on various configurations.
Issue id #1
##  Changes Made
- Added  feature engineering ( Outliers Remove, Scaling ) 
- hyperparameter tuning with Optuna .
- Added New algorithm such as CatBoostRegressor , Artificial Neural Network
- Refactored code for better readability and modularity.

### **Performance Comparison**
- ** Random Forest  **: 
  - old Accuracy: 87%
  - New Accuracy: 89%
- ** Gradient Boosting**:
  - old Accuracy: 87%
  - New Accuracy: 90%
- **CatBoostRegressor **:
  - old Accuracy:  -
  - New Accuracy: 90%
- **Artificial NN **:
  - old Accuracy:  -
  - New Accuracy: 93%

- Attached training and validation loss curves in the `results/` directory.

### CODE REVIEW

**Comment by garvit-010** [comment]
Nice job!! Thanks for your contribution
